kuma/core/middleware.py

```diff
@@ -44,7 +54,9 @@ def process_request(self, request):
             if query_string:
                 full_path = '%s?%s' % (full_path, query_string)
 
-            response = HttpResponsePermanentRedirect(full_path)
+            # Never use HttpResponsePermanentRedirect here.
+            # Its a temporary redirect and should return with http 302, not 301
+            response = HttpResponseRedirect(full_path)
```

Comment:
This change is needed. Suppose, the user choose `bn-BD` and we permanently redirect him to `bn-BD`, then the user chose to go `en-US` and as we permanently redirect them to `bn-BD`, the browser will always take them there. We can not do anything from our code end.